include/zephyr/bluetooth/iso.h

```diff
@@ -86,6 +86,18 @@ extern "C" {
 #define BT_ISO_BIS_INDEX_MIN        0x01
 /** Highest BIS index */
 #define BT_ISO_BIS_INDEX_MAX        0x1F
+/** Reset sequence number
+ *
+ * When this value is used as the sequence number in bt_iso_chan_send() the
+ * sequence number range starts again from 0.
+ */
+#define BT_ISO_SEQUENCE_NUMBER_RESET 0xFFFFU
```

Comment:
As the sequence number is a 16 bit unsigned value 0xffff is a valid sequence number to use (will occur every 11 minutes for 10ms frames).
Maybe use a 32 bit value for reset?
UPDATE: I see that the reset value is actually not used to force-reset to 0, but is accepted as a valid value - clever! - no change required:-)

Reply:
Hmm, one issue with this approach is the following scenario:

The host sends an SDU with SN `0xFFFE`, and the next SDU is 3 intervals later, and would thus be sent with SN `0x0001`. 

The approach with resetting the sequence number using `0xFFFF` won't work in that scenario. 

So I think that we either need to come up with some way of determining which SNs we would consider value (e.g. larger than current or perhaps lower than e.g. 1000), or we need to remove the verification of the sequence number all together and just send to the controller regardless.

Reply:
I think it would work.
If the app do not reset `0x0001` is larger than `0xFFFE` (as it wraps) - hence that is valid (if the check handles wraps of cause - I did not consider that during the review :-) )
If the app makes a call to reset the SN (`0xFFFF`) and then after 3 SDU intervals sends with the SN `0x0001` it should be accepted.

But yes for a uint16_t you would need to define what a larger value is. A difference of 1000 is a good number, but perhaps use half the range of uint16_t? But then again that basically disables the check and you might as well remove it. In the case you pause transmission for 6+ minutes the API call would fail even though you use a correct SN.

One alternative would be to have the API and check(iso_conn->iso.seq_num) use a 32 bit values - then you would not have the issue to determine whether the SN wrapped or not.

Reply:
> If the app makes a call to reset the SN (0xFFFF) and then after 3 SDU intervals sends with the SN 0x0001 it should be accepted.

The current implementation still uses 0xFFFF as a valid value, and would attempt to send a SDU, which would be unwanted in that case. To reset the counter without sending anything, we should need to use at least 17 bits and and have a "reset only" value. 

> But yes for a uint16_t you would need to define what a larger value is. A difference of 1000 is a good number, but perhaps use half the range of uint16_t? But then again that basically disables the check and you might as well remove it. In the case you pause transmission for 6+ minutes the API call would fail even though you use a correct SN.

When playing around with the updated shell approach, i.e. calculating the next SN instead of doing it by timer, it occurred to me how fast the SNs would wrap (approximately every 16 seconds at 255us). There is no way of handling sporadic sequence numbers from the shell as you can easily send SN 20000 and then the next be at SN 5000 when using something like the shell. Granted that the shell probably shouldn't be the thing that decides how we handle this, but on the other hand, given the issue with the SN for the peripheral and updating the SN every 255us, we still need something to work for the shell. 

> One alternative would be to have the API and check(iso_conn->iso.seq_num) use a 32 bit values - then you would not have the issue to determine whether the SN wrapped or not.

Until it's time to wrap the 32-bit value ;) But no, this could be a decent work around, as the wrapping timing would go from 65 seconds to 1193 hours (at 10ms interval), and we could then allow for values than the current once the current value reaches `0x10000` (i.e. larger than a uint16_t.